src/sentry/snuba/metrics/naming_layer/mapping.py

```diff
@@ -54,21 +54,21 @@ def get_mri(external_name: Union[Enum, str]) -> str:
         )
 
 
-def get_public_name_from_mri(internal_name: Union[TransactionMRI, SessionMRI, str]) -> str:
+def get_public_name_from_mri(
+    internal_name: Union[TransactionMRI, SessionMRI, str]
+) -> Optional[str]:
+    """Returns the public name from a MRI if its a builtin metric, None otherwise"""
     if not len(MRI_TO_NAME):
         create_name_mapping_layers()
 
     if isinstance(internal_name, Enum):
         internal_name = internal_name.value
     assert isinstance(internal_name, str)
 
-    try:
-        return MRI_TO_NAME[internal_name]
-    except KeyError:
-        raise InvalidParams(f"Unable to find a mri reverse mapping for '{internal_name}'.")
+    return MRI_TO_NAME.get(internal_name)
```

Comment:
I would prefer to not remove the logic that raises the `InvalidParams` exception but rather prior to raising the exception, we match the MRI against the custom namespace in the REGEX as mentioned here and if it matches the custom namespace then only then return a `None`